app/src/main/java/swati4star/createpdf/fragment/HomeFragment.java

```diff
@@ -528,4 +525,46 @@ private void resetValues() {
 
         return enhancementOptionsEntityArrayList;
     }
+
+    @Override
+    public void onItemClick(int position) {
+        switch (position) {
+            case 0:
+                passwordProtectPDF();
+                break;
+            case 1:
+                cropImages();
+                break;
+            default:
+                break;
+        }
+    }
+
+    private void passwordProtectPDF() {
+        if (mTempUris.size() == 0) {
+            Snackbar.make(Objects.requireNonNull(mActivity).findViewById(android.R.id.content),
+                    R.string.snackbar_no_images,
+                    Snackbar.LENGTH_LONG).show();
+            return;
+        }
+
+        new MaterialDialog.Builder(mActivity)
+                .title(R.string.set_password)
+                .content(R.string.enter_password)
+                .input(getString(R.string.example_password), null, new MaterialDialog.InputCallback() {
+                    @Override
+                    public void onInput(@NonNull MaterialDialog dialog, CharSequence input) {
+                        if (input == null || input.toString().trim().equals("")) {
```

Comment:
`input == null || input.toString().trim().equals("")`
This check is being used many times. Move this validation to a new class (`StringUtils`) in [util](https://github.com/Swati4star/Images-to-PDF/tree/master/app/src/main/java/swati4star/createpdf/util).